test-executor/src/main/java/com/cognifide/aet/executor/SuiteExecutor.java

```diff
@@ -0,0 +1,275 @@
+
+  private static final long DEFAULT_MESSAGE_RECEIVE_TIMEOUT = 20000L;
+
+  @Property(name = MESSAGE_RECEIVE_TIMEOUT_PROPERTY_NAME, label = "ActiveMQ message receive timeout",
+      description = "ActiveMQ message receive timeout", longValue = DEFAULT_MESSAGE_RECEIVE_TIMEOUT)
+  private Long messageReceiveTimeout;
+
+  private static final String CACHE_EXPIRATION_TIMEOUT_PROPERTY_NAME = "cacheExpirationTimeout";
+
+  private static final long DEFAULT_CACHE_EXPIRATION_TIMEOUT = 30000L;
+
+  @Property(name = CACHE_EXPIRATION_TIMEOUT_PROPERTY_NAME, label = "Suite runner cache expiration timeout",
+      description = "Suite runner cache expiration timeout. Should be longer than 'ActiveMQ message receive timeout'.",
+      longValue = DEFAULT_CACHE_EXPIRATION_TIMEOUT)
+  private Long cacheExpirationTimeout;
+
+  @Reference
+  private JmsConnection jmsConnection;
+
+  @Reference
+  private LockService lockService;
+
+  @Reference
+  private ReportConfigurationManager reportConfigurationManager;
+
+  private Cache<String, SuiteRunner> suiteRunnerCache;
+
+  private Cache<String, Queue<SuiteStatusResult>> suiteStatusCache;
+
+  private RunnerCacheUpdater runnerCacheUpdater;
+
+  private SuiteStatusHandler suiteStatusHandler;
+
+  @Activate
+  public void activate(Map properties) {
+    messageReceiveTimeout = PropertiesUtil.toLong(
+        properties.get(MESSAGE_RECEIVE_TIMEOUT_PROPERTY_NAME), DEFAULT_MESSAGE_RECEIVE_TIMEOUT);
+    cacheExpirationTimeout = PropertiesUtil.toLong(
+        properties.get(CACHE_EXPIRATION_TIMEOUT_PROPERTY_NAME), DEFAULT_CACHE_EXPIRATION_TIMEOUT);
+
+    suiteRunnerCache = CacheBuilder.newBuilder()
+        .expireAfterAccess(cacheExpirationTimeout, TimeUnit.MILLISECONDS)
+        .removalListener(new RunnerCacheRemovalListener())
+        .build();
+
+    suiteStatusCache = CacheBuilder.newBuilder()
+        .expireAfterAccess(cacheExpirationTimeout, TimeUnit.MILLISECONDS)
+        .build();
+
+    runnerCacheUpdater = new RunnerCacheUpdater(suiteRunnerCache);
+    suiteStatusHandler = new SuiteStatusHandler(suiteStatusCache);
+  }
+
+  /**
+   * Executes the test suite provided as a parameter.
+   *
+   * @param suiteString - content of the test suite XML file
+   * @param domain - overrides domain defined in the suite file
+   * @param endpointDomain - domain under which AET system is available, used to create links to
+   *                       xUnit report and status check
+   * @return status of the suite execution
+   */
+  public SuiteExecutionResult execute(String suiteString, String domain, String endpointDomain) {
+    SuiteRunner suiteRunner = null;
+    SuiteExecutionResult result;
+
+    TestSuiteParser xmlFileParser = new XmlTestSuiteParser();
+    try {
+      TestSuiteRun testSuiteRun = xmlFileParser.parse(suiteString);
+      testSuiteRun = overrideDomainIfDefined(testSuiteRun, domain);
+
+      String validationResult = validateTestSuiteRun(testSuiteRun);
```

Comment:
This class is getting big (> 200 llines), what do you think about extracting a Validator class for `testSuite` into some package-scoped class?
(nice to have)